bot/decorators.py

```diff
@@ -236,3 +238,31 @@ async def wrapped(*args, **kwargs) -> t.Any:
             return await func(*args, **kwargs)
         return wrapped
     return decorator
+
+
+def ensure_duration_in_future(duration_arg: function.Argument) -> t.Callable:
```

Comment:
How about `ensure_future_duration` as the name? It's slightly shorter.

I also humbly suggest to make this more generic (currently the name and error message tie it too close to infractions). The name could then be `ensure_future_timestamp`.

Reply:
resolved in 574adc3